### PR TITLE
New conditions

### PR DESCRIPTION
**Pull request by tarrox**
New functionality for erna. This PR enhances the current `process_fact_runs.py` and `fetch_fact_runs.py` to use the new @fact-project/fact_conditions project. This allows for setting nice and easy conditions for the data runs by specifying them in yaml files (see examples in fact_conditions project in the conditions folder) or by setting them directly in the cmd line or both. 

This change should work without any changes to current executions as long as it didn't use the `--conditions` option (which shouldn't have been used anyways as it does not do anything useful really).

Exp.:

`process_fact_data 20130630 20140206 /fact/raw ${FACT_TOOLS} ${XML_FOLDER}/example.xml /fact/aux ${output}/test.hdf5  --walltime=6:00:00 --num_runs 4 --vmem 10000 -c "@/home/projects/fact_tools/crab1314_moderate_moonlight.yaml`

`process_fact_data 20130630 20140206 /fact/raw ${FACT_TOOLS} ${XML_FOLDER}/example.xml /fact/aux ${output}/test.hdf5  --walltime=6:00:00 --num_runs 4 --vmem 10000 -c "@/home/projects/fact_tools/crab1314_moderate_moonlight.yaml" -c "fCurrentsMedMeanBeg > 64"`

### CODE REVIEW

File: erna/scripts/process_fact_mc.py [hunk 1]
```diff
@@ -14,23 +14,9 @@
 import glob
 
 logger = logging.getLogger(__name__)
-
-import re
-
-def create_filename_from_format(filename_format, basename, num):
```
Comment (jebuss) [inline erna/scripts/process_fact_mc.py]: Why is this deleted here? Didn't you add this function a couple of commits ago?

**Comment by jebuss** [comment]
please add a requirements.txt with the github requirements

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+git+https://github.com/fact-project/fact_conditions#egg=fact_conditions
+git+https://github.com/mackaiver/gridmap#egg=gridmap
```
Comment (MaxNoe) [inline requirements.txt]: Please use the form `https://github.com/fact-project/archive/<version>.tar.gz`, if there is not version, use `master` or make a release.

Installation is faster and we use a fixed version.

Comment (tarrox) [reply requirements.txt]: fact_conditions got a version

gridmap does nto have a tag or version so I gave it the commit hash so it is a fixed version

File: erna/scripts/process_fact_mc.py [hunk 3]
```diff
@@ -45,7 +31,7 @@ def make_jobs(jar, xml, data_paths, drs_paths,
         if output_path:
             # create the filenames for each single local run
             file_name, _ = path.splitext(path.basename(output_path))
-            file_name = create_filename_from_format(filename_format, file_name, num)
+            file_name = "{}_{}.{}".format(file_name, num, filename_format)
```
[the remark this answers is not in the document]
Comment (tarrox) [reply erna/scripts/process_fact_mc.py]: renamed `local_output_extension` to be more clear it is only for the local output

File: erna/scripts/process_fact_mc.py [hunk 4]
```diff
@@ -91,9 +77,7 @@ def make_jobs(jar, xml, data_paths, drs_paths,
               show_default=True)
 @click.option('--mcdrs', type=click.Path(exists=True, dir_okay=False, file_okay=True, readable=True))
 @click.option('--mcwildcard', help="Gives the wildcard for searching the folder for files.", type=click.STRING, default='**/*_Events.fit*')
-@click.option('--local_output_format', default="{basename}_{num}.json", help="Give the file format for the local output funktionality."
```
[the remark this answers is not in the document]
Comment (tarrox) [reply erna/scripts/process_fact_mc.py]: renamed `local_output_extension` to be more clear it is only for the local output

File: erna/scripts/process_fact_mc.py (hunk 1 above)
Comment (jebuss) [reply erna/scripts/process_fact_mc.py]: Anyways, just having the extension is fine with me.
